Build failed on Mac OS X with clang 3.5

Error message as below:

``` shell
[  0%] Building CXX object libopenage/CMakeFiles/libopenage.dir/game_control.cpp.o
In file included from /Users/dlab/Desktop/WTF/Git/openage/libopenage/game_control.cpp:3:
In file included from /Users/dlab/Desktop/WTF/Git/openage/libopenage/gamestate/game_spec.h:5:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/unordered_map:350:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/__hash_table:16:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/memory:602:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/utility:259:9: error: field has incomplete type 'const openage::Command'
    _T2 second;
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/deque:913:49: note: in instantiation of template class
      'std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command>' requested here
    static const difference_type __block_size = sizeof(value_type) < 256 ? 4096 / sizeof(value_type) : 16;
                                                ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/deque:1184:15: note: in instantiation of template class
      'std::__1::__deque_base<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command>,
      std::__1::allocator<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command> > >' requested here
    : private __deque_base<_Tp, _Allocator>
              ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/queue:197:22: note: in instantiation of template class
      'std::__1::deque<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command>,
      std::__1::allocator<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command> > >' requested here
    typedef typename container_type::value_type      value_type;
                     ^
/Users/dlab/Desktop/WTF/Git/openage/libopenage/gamestate/../unit/unit.h:259:69: note: in instantiation of template class
      'std::__1::queue<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command>,
      std::__1::deque<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command>,
      std::__1::allocator<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command> > > >' requested here
        std::queue<std::pair<std::shared_ptr<UnitAbility>, const Command>> command_queue;
                                                                           ^
/Users/dlab/Desktop/WTF/Git/openage/libopenage/gamestate/../unit/unit_container.h:13:7: note: forward declaration of 'openage::Command'
class Command;
      ^
In file included from /Users/dlab/Desktop/WTF/Git/openage/libopenage/game_control.cpp:5:
In file included from /Users/dlab/Desktop/WTF/Git/openage/libopenage/game_control.h:6:
In file included from /Users/dlab/Desktop/WTF/Git/openage/libopenage/input/input_context.h:6:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/stack:86:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/deque:1194:30: error: '__alloc_traits' is a protected member of
      'std::__1::__deque_base<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command>,
      std::__1::allocator<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command> > >'
    typedef typename __base::__alloc_traits        __alloc_traits;
                             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/queue:197:22: note: in instantiation of template class
      'std::__1::deque<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command>,
      std::__1::allocator<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command> > >' requested here
    typedef typename container_type::value_type      value_type;
                     ^
/Users/dlab/Desktop/WTF/Git/openage/libopenage/gamestate/../unit/unit.h:259:69: note: in instantiation of template class
      'std::__1::queue<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command>,
      std::__1::deque<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command>,
      std::__1::allocator<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command> > > >' requested here
        std::queue<std::pair<std::shared_ptr<UnitAbility>, const Command>> command_queue;
                                                                           ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/deque:905:54: note: declared protected here
    typedef allocator_traits<allocator_type>         __alloc_traits;
                                                     ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/deque:1195:30: error: 'reference' is a protected member of
      'std::__1::__deque_base<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command>,
      std::__1::allocator<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command> > >'
    typedef typename __base::reference             reference;
                             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/deque:906:54: note: declared protected here
    typedef value_type&                              reference;
                                                     ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/deque:1196:30: error: 'const_reference' is a protected member of
      'std::__1::__deque_base<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command>,
      std::__1::allocator<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command> > >'
    typedef typename __base::const_reference       const_reference;
                             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/deque:907:54: note: declared protected here
    typedef const value_type&                        const_reference;
                                                     ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/deque:1197:30: error: 'iterator' is a protected member of
      'std::__1::__deque_base<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command>,
      std::__1::allocator<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command> > >'
    typedef typename __base::iterator              iterator;
                             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/deque:935:64: note: declared protected here
                             difference_type, __block_size>    iterator;
                                                               ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/deque:1198:30: error: 'const_iterator' is a protected member of
      'std::__1::__deque_base<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command>,
      std::__1::allocator<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command> > >'
    typedef typename __base::const_iterator        const_iterator;
                             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/deque:937:64: note: declared protected here
                             difference_type, __block_size>    const_iterator;
                                                               ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/deque:1199:30: error: 'size_type' is a protected member of
      'std::__1::__deque_base<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command>,
      std::__1::allocator<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command> > >'
    typedef typename __base::size_type             size_type;
                             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/deque:908:54: note: declared protected here
    typedef typename __alloc_traits::size_type       size_type;
                                                     ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/deque:1200:30: error: 'difference_type' is a protected member of
      'std::__1::__deque_base<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command>,
      std::__1::allocator<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command> > >'
    typedef typename __base::difference_type       difference_type;
                             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/deque:909:54: note: declared protected here
    typedef typename __alloc_traits::difference_type difference_type;
                                                     ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/deque:1202:30: error: 'pointer' is a protected member of
      'std::__1::__deque_base<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command>,
      std::__1::allocator<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command> > >'
    typedef typename __base::pointer               pointer;
                             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/deque:910:54: note: declared protected here
    typedef typename __alloc_traits::pointer         pointer;
                                                     ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/deque:1203:30: error: 'const_pointer' is a protected member of
      'std::__1::__deque_base<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command>,
      std::__1::allocator<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command> > >'
    typedef typename __base::const_pointer         const_pointer;
                             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/deque:911:54: note: declared protected here
    typedef typename __alloc_traits::const_pointer   const_pointer;
                                                     ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/deque:1364:30: error: '__map_const_pointer' is a protected member of
      'std::__1::__deque_base<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command>,
      std::__1::allocator<std::__1::pair<std::__1::shared_ptr<openage::UnitAbility>, const openage::Command> > >'
    typedef typename __base::__map_const_pointer __map_const_pointer;
                             ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/deque:931:81: note: declared protected here
    typedef typename allocator_traits<__const_pointer_allocator>::const_pointer __map_const_pointer;
                                                                                ^
11 errors generated.
make[3]: *** [libopenage/CMakeFiles/libopenage.dir/game_control.cpp.o] Error 1
make[2]: *** [libopenage/CMakeFiles/libopenage.dir/all] Error 2
```

Adding `#include "command.h"` to unit/unit.h helps nothing.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/32574632-build-failed-on-mac-os-x-with-clang-3-5?utm_campaign=plugin&utm_content=tracker%2F371436&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F371436&utm_medium=issues&utm_source=github).
</bountysource-plugin>

I do have another problem with clang 3.8 (not apple clang!) in debug mode (release works fine) with a line in ```font.cpp```
```c++
// TODO Replace this trickery with hb_ft_font_create_referenced for harfbuzz version > 1.0
FT_Reference_Face(ft_face);
```
Error message
```c++
font.cpp:121:2: error: use of undeclared identifier 'FT_Reference_Face'
```
Changing this to
```c++
hb_ft_font_create_referenced(ft_face);
```
works fine and the harfbuzz version should be 1.x.x, at least this is the stable version under active maintenance.

Can anyone confirm this?

--- a/copying.md
+++ b/copying.md
@@ -78,6 +78,7 @@ _the openage authors_ are:
 | Guillaume Desquesnes        | elnabo                      | g.desquesnes@gmail.com                |
 | Johan Klokkhammer Helsing   | johanhelsing                | johanhelsing@gmail.com                |
 | Jasper v. Blanckenburg      | jazzpi                      | jasper@mezzo.de                       |
+| Alexej Disterhoft           | nobbs                       | disterhoft@uni-mainz.de               |
 
 If you're a first-time commiter, add yourself to the above list. This is not
 just for legal reasons, but also to keep an overview of all those nicknames.

--- a/libopenage/unit/ability.cpp
+++ b/libopenage/unit/ability.cpp
@@ -3,6 +3,7 @@
 #include <memory>
 
 #include "../terrain/terrain_object.h"
+#include "../gamestate/player.h"
 #include "ability.h"
 #include "action.h"
 #include "command.h"

--- a/libopenage/unit/command.h
+++ b/libopenage/unit/command.h
@@ -6,8 +6,6 @@
 
 #include "../coord/phys3.h"
 #include "ability.h"
-#include "producer.h"
-#include "unit.h"
 
 namespace openage {
 
@@ -42,6 +40,8 @@ struct hash<openage::command_flag> {
 namespace openage {
 
 class Player;
+class Unit;
+class UnitType;
 
 /*
  * Game command from the ui

--- a/libopenage/unit/unit.h
+++ b/libopenage/unit/unit.h
@@ -14,6 +14,7 @@
 #include "../handlers.h"
 #include "ability.h"
 #include "attribute.h"
+#include "command.h"
 #include "unit_container.h"
 
 namespace openage {

